crates/transaction-pool/src/pool/pending.rs

```diff
@@ -150,6 +150,51 @@ impl<T: TransactionOrdering> PendingPool<T> {
         self.by_id.values().map(|tx| tx.transaction.clone())
     }
 
+    /// Updates the pool with the new blob fee. Removes
+    /// from the subpool all transactions and their dependents that no longer satisfy the given
+    /// base fee (`tx.max_blob_fee < blob_fee`).
+    ///
+    /// Note: the transactions are not returned in a particular order.
+    ///
+    /// # Returns
+    ///
+    /// Removed transactions that no longer satisfy the blob fee.
+    pub(crate) fn update_blob_fee(
+        &mut self,
+        blob_fee: u128,
+    ) -> Vec<Arc<ValidPoolTransaction<T::Transaction>>> {
```

Comment:
hmm, I wonder if we can do blobfee and basefee checks in a single function.
since this function is identical to the basefee update

though, both can move in different direction.
we could add some helper that knows which checks to do.

but maybe it's better to keep this separate for now